interfaces/utils/utils.go

```diff
@@ -40,3 +45,40 @@ func NormalizeInterfaceAttributes(value interface{}) interface{} {
 	}
 	return value
 }
+
+// CopyAttributes makes a deep copy of the attributes map.
+func CopyAttributes(value map[string]interface{}) map[string]interface{} {
+	return copyRecursive(value).(map[string]interface{})
+}
+
+func copyRecursive(value interface{}) interface{} {
+	// note: ensure all the mutable types (or types that need a conversion)
+	// are handled here.
+	switch v := value.(type) {
+	case []interface{}:
+		arr := make([]interface{}, len(v))
+		for i, el := range v {
+			arr[i] = copyRecursive(el)
+		}
+		return arr
+	case map[string]interface{}:
+		mp := make(map[string]interface{}, len(v))
+		for key, item := range v {
+			mp[key] = copyRecursive(item)
+		}
+		return mp
+	}
+	return value
+}
+
+// Regular expression describing correct identifiers.
+var validName = regexp.MustCompile("^[a-z](?:-?[a-z0-9])*$")
+
+// ValidateName checks if a string can be used as a plug or slot name.
+func ValidateName(name string) error {
```

Comment:
Please drop this and look at `snap.Validate{Plug,Slot}Name` instead.

Reply:
Great idea, thanks. Done.